Tasks/AppStoreRelease/app-store-release.ts

```diff
@@ -270,39 +272,49 @@ async function run() {
             let bundleIdentifier: string = tl.getInput('appIdentifier', false);
+            let externalTestersGroups: string = tl.getInput('externalTestersGroups');
+            let authArgs: string[];
             if (isUsingApiKey) {
-                pilotCommand.arg(['pilot', 'upload', '--api_key_path', apiKeyFilePath, '-i', filePath]);
+                authArgs = ['--api_key_path', apiKeyFilePath];
             } else {
-                pilotCommand.arg(['pilot', 'upload', '-u', credentials.username, '-i', filePath]);
+                authArgs = ['-u', credentials.username];
             }
-            let usingReleaseNotes: boolean = isValidFilePath(releaseNotes);
-            if (usingReleaseNotes) {
-                if (!credentials.fastlaneSession) {
-                    tl.warning(tl.loc('ReleaseNotesRequiresFastlaneSession'));
-                }
+            if (distributeOnly) {
+                pilotCommand.arg(['pilot', 'distribute', ...authArgs]);
+                pilotCommand.argIf(appBuildNumber, ['--build_number', appBuildNumber]);
+                pilotCommand.argIf(bundleIdentifier, ['-a', bundleIdentifier]);
+                pilotCommand.argIf(externalTestersGroups, ['--groups', externalTestersGroups]);
+            } else {
+                pilotCommand.arg(['pilot', 'upload', ...authArgs]);
+                pilotCommand.arg(['-i', filePath]);
+                let usingReleaseNotes: boolean = isValidFilePath(releaseNotes);
+                if (usingReleaseNotes) {
+                    if (!credentials.fastlaneSession) {
+                        tl.warning(tl.loc('ReleaseNotesRequiresFastlaneSession'));
+                    }
 
-                pilotCommand.arg(['--changelog', fs.readFileSync(releaseNotes).toString()]);
-            }
-            pilotCommand.argIf(teamId, ['-q', teamId]);
-            pilotCommand.argIf(teamName, ['-r', teamName]);
-            pilotCommand.argIf(bundleIdentifier, ['-a', bundleIdentifier]);
-            pilotCommand.argIf(shouldSkipSubmission, ['--skip_submission', 'true']);
-            pilotCommand.argIf(shouldSkipWaitingForProcessing, ['--skip_waiting_for_build_processing', 'true']);
-            pilotCommand.argIf(appSpecificId, ['-p', appSpecificId]);
-
-            let distributedToExternalTesters: boolean = tl.getBoolInput('distributedToExternalTesters', false);
-            if (distributedToExternalTesters) {
-                tl.debug('Distributing to external testers');
-                if (!usingReleaseNotes) {
-                    throw new Error(tl.loc('ReleaseNotesRequiredForExternalTesting'));
-                }
-                pilotCommand.arg(['--distribute_external', 'true']);
-                if (shouldSkipSubmission || shouldSkipWaitingForProcessing) {
-                    tl.warning(tl.loc('ExternalTestersCannotSkipWarning'));
+                    pilotCommand.arg(['--changelog', fs.readFileSync(releaseNotes).toString()]);
```

Comment:
Is changelog actual for distribute commands also? Could you please re-check which args are also actual for both commands/which inputs could we add for distribute (maybe in separate ticket)?

Reply:
Yes. The changelog is actual for distribute command. In this case, it will replace the previously uploaded text. Not sure if this is needed now, but I can add it to this PR.
Created a work item to investigate the rest of the arguments: https://github.com/microsoft/build-task-team/issues/1163